common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -1123,6 +1181,8 @@ public CollationIdentifier identifier() {
   public static final List<String> SUPPORTED_PROVIDERS = List.of(PROVIDER_SPARK, PROVIDER_ICU);
   public static final String COLLATION_PAD_ATTRIBUTE = "NO_PAD";
 
+  public static final int DEFAULT_COLLATION_ID =
```

Comment:
I feel like this being public can introduce bugs in the future because someone just checks the id and doesn't call the function; can we make it private or remove it altogether?

Reply:
Yes, we can, in fact, I changed the logic ini all places of code that compares `collationId` directly. I think this single case will not be a problem since the problematic places in the code are the ones that compare `collationId` directly. I changed those since we can compare the `StringType` directly without accessing its `collationId`. So, I do not think that this one alone will be a problem, but rather comparisons of `collationId` directly.